src/ngraph/runtime/cpu/cpu_backend.cpp

```diff
@@ -32,6 +35,55 @@ std::shared_ptr<ngraph::runtime::TensorView>
     runtime::cpu::CPU_Backend::make_primary_tensor_view(const ngraph::element::Type& element_type,
                                                         const Shape& shape)
 {
-    auto rc = make_shared<runtime::cpu::CPUTensorView>(element_type, shape);
-    return dynamic_pointer_cast<runtime::TensorView>(rc);
+    return make_shared<runtime::cpu::CPUTensorView>(element_type, shape);
+}
+
+std::shared_ptr<ngraph::runtime::TensorView>
+    runtime::cpu::CPU_Backend::create_tensor(const ngraph::element::Type& element_type,
+                                             const Shape& shape)
+{
+    return make_shared<runtime::cpu::CPUTensorView>(element_type, shape);
+}
+
+bool runtime::cpu::CPU_Backend::compile(const ngraph::Function& func)
+{
+    m_function = clone_function(func);
+    if (m_external_function)
+    {
+        throw runtime_error("Backend can only compile a single function");
+    }
+    m_external_function = make_shared<cpu::CPU_ExternalFunction>(m_function);
+    auto cf = m_external_function->make_call_frame();
+    m_call_frame = dynamic_pointer_cast<cpu::CPU_CallFrame>(cf);
+    return true;
+}
+
+bool runtime::cpu::CPU_Backend::call(const Function& fun,
+                                     const vector<shared_ptr<runtime::TensorView>>& outputs,
+                                     const vector<shared_ptr<runtime::TensorView>>& inputs)
+{
+    bool rc = true;
+    try
+    {
+        compile(fun);
+        call(outputs, inputs);
+    }
+    catch (...)
```

Comment:
This will change the expectations of a couple of unit tests (specifically the one_hot tests, and a couple others I can't remember) which are expecting to catch certain exceptions thrown inside the generated code. But in retrospect perhaps that's not something we should be doing in the first place.